# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../llvm/include-paths.patch
# Copyright (C) 2009 - 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- llvm-2.9//utils/clang/lib/Frontend/InitHeaderSearch.cpp.orig	2011-03-22 01:24:04.000000000 +0100
+++ llvm-2.9//utils/clang/lib/Frontend/InitHeaderSearch.cpp	2011-04-08 17:25:09.000000000 +0200
@@ -450,6 +450,8 @@
     break;
   }
 
+  AddPath("@T2SDE_SYSTEM_INCLUDE_PATH@", System, false, false, false);
+
   // Builtin includes use #include_next directives and should be positioned
   // just prior C include dirs.
   if (HSOpts.UseBuiltinIncludes) {
@@ -639,6 +641,11 @@
     AddPath("/usr/include/c++/4.1", CXXSystem, true, false, false);
     break;
   case llvm::Triple::Linux:
+
+    AddGnuCPlusPlusIncludePaths("@T2SDE_CXX_SYSTEM_INCLUDE_PATH@",
+                                "@T2SDE_ARCH_TARGET@", "@T2SDE_DIR32@", "@T2SDE_DIR64@", triple);
+
+#if 0
     //===------------------------------------------------------------------===//
     // Debian based distros.
     // Note: these distros symlink /usr/include/c++/X.Y.Z -> X.Y
@@ -815,7 +822,7 @@
     AddGnuCPlusPlusIncludePaths(
         "/usr/lib/llvm-gcc-4.2-9999/include/c++/4.2.1",
         "x86_64-pc-linux-gnu", "", "", triple);
-
+#endif
     break;
   case llvm::Triple::FreeBSD:
     // FreeBSD 8.0
